packages/core/src/threatComposer/messageHandlers/addFileWatchMessageHandler.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Command, FileChangedMessage, MessageType, WebviewContext } from '../types'
+import vscode from 'vscode'
+import { fsCommon } from '../../srcShared/fs'
+
+/**
+ * Function to add a watcher on the file that was opened. The watcher will notify Threat Composer
+ * view when a change occurs to the file externally.
+ * If the change occurs due to
+ *    a user save action, file change is ignored.
+ *    an auto save that is persisted as AutoSave is turned on in VSCode settings, update file state.
+ *    an external file change,
+ *       if a local, unsaved change exist in Threat Composer, the file change is ignored.
+ *       if no unsaved local changes exist, the view is notified of the external file change.
+ * @param context: The Webview Context that contain the details of the file and the webview
+ */
+export function addFileWatchMessageHandler(context: WebviewContext) {
```

Comment:
why not just call it `onDidChangeTextDocument` (or `onFileChanged`) ? "on" is the conventional prefix, please follow the codebase conventions.